pkgs/development/python-modules/numba/default.nix

```diff
@@ -70,18 +79,35 @@ in buildPythonPackage rec {
     done
   '';
 
-  # Copy test script into $out and run the test suite.
+  # run a smoke test in a temporary directory so that
+  # a) Python picks up the installed library in $out instead of the build files
+  # b) we have somewhere to put $HOME so some caching tests work
+  # c) it doesn't take 6 CPU hours for the full suite
   checkPhase = ''
-    ${python.interpreter} -m numba.runtests
+    pushd $(mktemp -d)
+    HOME=. ${python.interpreter} -m numba.runtests -m $NIX_BUILD_CORES numba.tests.test_usecases
+    popd
   '';
 
-  # ImportError: cannot import name '_typeconv'
-  doCheck = false;
-
   pythonImportsCheck = [
     "numba"
   ];
 
+  passthru.tests = {
+    # CONTRIBUTOR NOTE: numba also contains CUDA tests, though these cannot be run in
+    # this sandbox environment. Consider running similar commands to those below outside the
+    # sandbox manually if you have the appropriate hardware; support will be detected
+    # and the corresponding tests enabled automatically.
+    fullSuite = runCommand "${pname}-test" {} ''
+      pushd $(mktemp -d)
+      # pip and python in $PATH is needed for the test suite to pass fully
+      PATH=${python.withPackages (p: [ p.numba p.pip ])}/bin:$PATH
+      HOME=. python -m numba.runtests -m $NIX_BUILD_CORES
+      popd
+      touch $out # stop Nix from complaining no output was generated and failing the build
+    '';
```

Comment:
If this is impure anyway we might as well write the command to $out/bin/ and run it without the sandbox.

Reply:
Are maintainers expected to interact with the packages listed in `passthru.tests` to test them or just be sure they build? The only impurity here is if the CUDA tests will run or not.

Reply:
The manual says about `passthru.tests` "An attribute set with tests as values. A test is a derivation that builds when the test passes and fails to build otherwise.". So it should be left as-is where building it runs the test.

Reply:
Yeah there's a danger someone will blindly go "yup, `passthru.tests` builds" and not realize that it slightly breaks the normal protocol.

Reply:
Ofborg will build those to make sure there are no downstream regressions.